tests/content-block-manager.spec.js

```diff
@@ -0,0 +1,133 @@
+import { expect } from "@playwright/test";
+import { test } from "../lib/cachebust-test";
+import { publishingAppUrl, waitForUrlToBeAvailable } from "../lib/utils";
+
+test.describe("Content Block Manager", { tag: ["@app-content-object-store"] }, () => {
+  let embedCode;
+  let value;
+
+  const contentBlockPath = `${publishingAppUrl("whitehall-admin")}/content-block-manager/content-block/18`;
+  const whitehallPath = "/government/admin/news/1658299";
+  const mainstreamPath = "/editions/663e33d76187a0001afaf022";
+
+  test.beforeAll(async ({ page }) => {
+    await page.goto(contentBlockPath);
+
+    const row = page.getByTestId("rate_1_amount");
+    value = await row.locator(".govuk-summary-list__value").first().textContent();
+    embedCode = await row.getAttribute("data-embed-code");
+  });
+
+  test.describe("When embedding content in Whitehall", () => {
+    test.use({ baseURL: publishingAppUrl("whitehall-admin") });
+
+    test("Can embed content in Whitehall", async ({ page }) => {
+      await page.goto(whitehallPath);
+      await page.getByRole("button", { name: "Edit draft" }).click();
+
+      await page.getByLabel("Body (required)").fill("");
+      await page.getByLabel("Body (required)").fill(embedCode);
+
+      await page.getByRole("button", { name: "Save and go to document" }).click();
+
+      const url = await waitForUrlToBeAvailable(
+        page,
+        await page.getByRole("link", { name: "Preview on website" }).getAttribute("href")
+      );
+
+      await page.goto(url);
+      await expect(page.getByText(value)).toBeVisible();
+    });
+  });
+
+  test.describe("When embedding content in Mainstream", () => {
+    test.use({ baseURL: publishingAppUrl("publisher") });
+
+    test("Can embed content in Mainstream", async ({ page }) => {
+      await page.goto(mainstreamPath);
+
+      await page.getByLabel("More information").fill("");
+      await page.getByLabel("More information").fill(embedCode);
+
+      await page.getByRole("button", { name: "Save" }).click();
+
+      const url = await waitForUrlToBeAvailable(
+        page,
+        await page.getByRole("link", { name: "Preview" }).getAttribute("href")
+      );
+
+      await page.goto(url);
+      await expect(page.getByText(value)).toBeVisible();
+    });
+  });
+
+  test.describe("When content changes", () => {
+    let newValue;
+
+    test.beforeAll(async ({ page }) => {
+      // Set a new random value between £100 and £200
+      newValue = `£${(Math.random() * (100.0 - 200.0) + 200.0).toFixed(2)}`;
+      await page.goto(contentBlockPath);
+
+      await page.getByRole("button", { name: "Edit pension" }).click();
+
+      await page.getByRole("button", { name: "Save and continue" }).click();
+
+      await page.locator('[data-test-id="embedded_rate-1"]').getByRole("link", { name: "Edit Rate details" }).click();
```

Comment:
Are you use the link is correct? 
<img width="1220" alt="Screenshot 2025-03-13 at 13 31 51" src="https://github.com/user-attachments/assets/cbc460af-8285-41ee-9788-9addfce1e76d" />

Reply:
Yeah, the "Rate details" part is in a `visually-hidden` span, but I can probably remove that bit from the tests, as Playwright doesn't need all the text

Reply:
oh yeah. As you can see in the screenshot I haven't opened that accordion - so all good! 😁

Reply:
👍